### PR TITLE
Remake events

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ✔️
| Fixed issues  | #79 #43

- Split events to View events and WebView events
- Add event `Head`
- Pass parameters to all events: `BeforeRender` and `AfterRender` use parameters of view, rest events use parameters passed to mark methods (`beginPage()`, `beginBody()` and other)
- Pass context object (`View` of `WebView`) to all events

### CODE REVIEW

**Review by devanych — APPROVED** [review]

**Review by terabytesoftw — APPROVED** [review]

**Comment by samdark** [comment]
What problem each of these changes solve?

**Comment by vjik** [comment]
Add event Head: https://github.com/yiisoft/view/issues/43

Pass context object (View of WebView) to all events: https://github.com/yiisoft/view/issues/79

Split events to View events and WebView events: events has different data (View or WebView). When events splitted we have right types.

Pass parameters to all events: BeforeRender and AfterRender use parameters of view, rest events use parameters passed to mark methods (beginPage(), beginBody() and other): In BeforeRender and AfterRender we have all parameters from view, in rest events we don't have parameters (this fire from view template) and can only send parameters to mark functions (beginPage(), beginBody() and other).

File: src/View.php [hunk 1]
```diff
@@ -17,22 +21,36 @@
...
-        $this->eventDispatcher->dispatch(new PageBegin($this->getViewFile()));
+        $this->eventDispatcher->dispatch(new PageBegin($this, $parameters));
     }
 
     /**
-     * Marks the ending of a page.
+     * Marks the ending of a view.
      */
-    public function endPage(): void
+    public function endPage(array $parameters = []): void
```
Comment (samdark) [inline src/View.php:37]: I don't think parameters are useful here and in alike methods.

Comment (vjik) [reply src/View.php]: Done

**Review by samdark — APPROVED** [review]

**Pull request merged**